lib/Dialect/Arc/Transforms/GroupResetsAndEnables.cpp

```diff
@@ -114,6 +114,68 @@ struct EnableGroupingPattern : public OpRewritePattern<ClockTreeOp> {
     return success(changed);
   }
 };
+
+struct GroupAssignmentsInIfPattern : public OpRewritePattern<ClockTreeOp> {
```

Comment:
Have you thought of letting the pattern match on `scf.if` directly and in the pattern check that it has an ancestor of type `ClockTreeOp` (e.g., using `getParentOfType<ClockTreeOp>()`)? Then you probably don't need to do all the iterations to collect the if-ops. I think this pattern is very useful also for if-ops not specifically created for the reset/enable grouping.

Reply:
Agreed, I'll make this change!